sonar-orchestrator/src/test/java/com/sonar/orchestrator/OrchestratorBuilderTest.java

```diff
@@ -237,42 +237,89 @@ public void use_default_admin_credentials_for_builds_if_enabled() {
 
       Build<FakeBuild> fakeBuild = FakeBuild.create(successResult);
       orchestrator.executeBuild(fakeBuild);
-      String adminToken = fakeBuild.getProperties().get("sonar.login");
-      assertThat(adminToken).isNotBlank();
+      String adminToken = verifyTokenCreated(fakeBuild, "sonar.token");
 
-      //second execution should reuse admin token
       fakeBuild = FakeBuild.create(successResult);
       orchestrator.executeBuild(fakeBuild);
-      String secondAdminToken = fakeBuild.getProperties().get("sonar.login");
-      assertThat(secondAdminToken).isNotBlank()
-        .isEqualTo(adminToken);
+      verifyTokenReused(fakeBuild, "sonar.token", adminToken);
 
       fakeBuild = FakeBuild.create(successResult);
       fakeBuild.setProperty("sonar.login", "custom-property-token");
       orchestrator.executeBuildQuietly(fakeBuild);
+      verifyTokenFromProperty(fakeBuild, "sonar.login", "custom-property-token");
```

Comment:
That is not exactly what I had in mind, I was more thinking of having the whole bloc inside the method, so just read the test like this :
````
verifyTokenCreated(..);
verifyTokenReused(..);
verifyTokentFromProperty(..);
...
verifyMultipleBuilds(..);
````
But in the end, we still got rid of the comment. 
So, no strong opinion for me on this.

Reply:
You are right that's better, I will update